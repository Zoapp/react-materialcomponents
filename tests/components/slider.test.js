import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Slider from "libs/components/slider";

describe("components/Slider", () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<Slider />);
  });

  it("can be discrete", () => {
    wrapper.setProps({ discrete: true });
    expect(wrapper.hasClass("mdc-slider--discrete")).toEqual(true);
  });

  it("can be disabled", () => {
    wrapper.setProps({ disabled: true });
    expect(wrapper.hasClass("mdc-slider--disabled")).toEqual(true);
  });

  it("can have a presetted min value", () => {
    wrapper.setProps({ min: 0 });
    expect(wrapper.prop("aria-valuemin")).toEqual(0);
  });

  it("can have a presetted max value", () => {
    wrapper.setProps({ max: 100 });
    expect(wrapper.prop("aria-valuemax")).toEqual(100);
  });

  it("can have a presetted value", () => {
    wrapper.setProps({ value: 50 });
    expect(wrapper.prop("aria-valuenow")).toEqual(50);
  });

  it("can have a label", () => {
    wrapper.setProps({ label: "foo" });
    expect(wrapper.prop("aria-label")).toEqual("foo");
  });

  it("can have a display marker", () => {
    const component = renderer.create(<Slider displayMarker />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("onInput()", () => {
  it("should call onInput callback", () => {
    const onInputSpy = jest.fn();
    const wrapper = shallow(<Slider onInput={onInputSpy} />);

    wrapper.instance().moveThumbPosition = jest.fn().mockReturnValue("value");
    wrapper.instance().handleMoveEnd({ preventDefault: () => {} });
    expect(onInputSpy).toHaveBeenCalledWith("value");
  });
});

describe("onChange()", () => {
  it("should call onChange callback", () => {
    const onChangeSpy = jest.fn();
    const wrapper = shallow(<Slider disabled={false} onChange={onChangeSpy} />);

    wrapper.setState({ active: true });
    wrapper.instance().moveThumbPosition = jest.fn().mockReturnValue("value");
    wrapper.instance().handleMoveChange({ preventDefault: () => {} });
    expect(onChangeSpy).toHaveBeenCalledWith("value");
  });
});
