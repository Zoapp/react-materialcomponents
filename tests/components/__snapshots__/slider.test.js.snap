// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Slider can be disabled 1`] = `
<div
  aria-disabled="true"
  aria-label=""
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="mdc-slider mdc-slider--disabled"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
      style={undefined}
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`components/Slider can be discrete 1`] = `
<div
  aria-label=""
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="mdc-slider mdc-slider--discrete"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
      style={undefined}
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <div
      className="mdc-slider__pin"
    >
      <span
        className="mdc-slider__pin-value-marker"
      >
        0
      </span>
    </div>
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`components/Slider can have a display marker 1`] = `
<div
  aria-label=""
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="mdc-slider"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
      style={undefined}
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`components/Slider can have a label 1`] = `
<div
  aria-label="foo"
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="mdc-slider"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
      style={undefined}
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`components/Slider can have a presetted max value 1`] = `
<div
  aria-label=""
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="mdc-slider"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
      style={undefined}
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`components/Slider can have a presetted min value 1`] = `
<div
  aria-label=""
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={0}
  className="mdc-slider"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
      style={undefined}
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;

exports[`components/Slider can have a presetted value 1`] = `
<div
  aria-label=""
  aria-valuemax={100}
  aria-valuemin={0}
  aria-valuenow={50}
  className="mdc-slider"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="slider"
  tabIndex="0"
>
  <div
    className="mdc-slider__track-container"
  >
    <div
      className="mdc-slider__track"
      style={undefined}
    />
  </div>
  <div
    className="mdc-slider__thumb-container"
  >
    <svg
      className="mdc-slider__thumb"
      height="21"
      width="21"
    >
      <circle
        cx="10.5"
        cy="10.5"
        r="7.875"
      />
    </svg>
    <div
      className="mdc-slider__focus-ring"
    />
  </div>
</div>
`;
